src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -414,6 +426,38 @@ public void SniffIncomingMessage(Message message)
             }
         }
 
+        private static void UnwrapCancellationToken(IAddressable target, object arg, InvokeMethodRequest request, int i)
+        {
+            // essentially unwrapping tokens that were wrapped before request sending
+            var orleansTokenWrapper = ((CancellationTokenWrapper)arg);
+            if (orleansTokenWrapper.WentThroughSerialization
```

Comment:
`WentThroughSerialization` exists for optimization in case of local calls - if no serialization happened - grain extension does not being created, and the token just being unwrapped.